src/lib/cache/subCache.cpp

```diff
@@ -720,6 +720,16 @@ void subCacheItemInsert
   cSubP->expression.coords     = coords;
   cSubP->expression.georel     = georel;
 
+  std::string errorString;
+  if (q != "")
+  {
+    if (cSubP->expression.stringFilter.parse(q.c_str(), &errorString) == false)
+    {
+      LM_E(("Runtime Error (error parsing q-string '%s': %s)", q.c_str(), errorString.c_str()));
```

Comment:
In addition, is the alarm logic "tied" with this error condition detection? I mean, would an error in q filter at subscription creation/updating time to raise a bad input alarm?